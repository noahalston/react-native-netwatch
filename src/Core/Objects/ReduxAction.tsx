import { ILog, LogType } from '../../types';

interface IAction {
  type: string;
  payload: any;
}

export class ReduxAction implements ILog {
  _id: number = -1;
  startTime: number = Date.now();
  type: LogType = 'REDUX';
  action: IAction = { type: '__ERROR:UNDEFINED__', payload: '' };

  constructor(attributes?: any) {
    Object.assign(this, attributes);
  }
}

export default ReduxAction;
