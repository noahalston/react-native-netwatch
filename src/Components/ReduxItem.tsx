import * as React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import ReduxAction from '../Core/Objects/ReduxAction';
import { getTime } from '../Utils/helpers';
import { Status } from './Status';

export interface IProps {
  item: ReduxAction;
}

export const ReduxItem: React.FC<IProps> = ({ item }) => (
  <View style={styles.listItemContainer}>
    <Status item={item} backgroundColor="#64b5f6" text="ACT" subText="-" />
    <View style={styles.url}>
      <View style={styles.action}>
        <Text numberOfLines={1} style={styles.titleStyle}>{`${
          item.action.type.toUpperCase() || 'UNDEFINED'
        }`}</Text>
        <Text style={styles.date}>{getTime(item.startTime)}</Text>
      </View>
      <Text numberOfLines={1} style={styles.descriptionStyle}>
        Redux action
      </Text>
    </View>
  </View>
);

export default ReduxItem;

const styles = StyleSheet.create({
  listItemContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    borderBottomColor: '#bdbdbd',
    borderBottomWidth: 0.5,
    backgroundColor: '#757575',
    height: 50,
  },
  url: {
    flex: 1,
    paddingRight: 8,
  },
  action: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  date: {
    color: 'white',
  },
  titleStyle: {
    paddingLeft: 16,
    color: 'white',
    fontWeight: 'bold',
  },
  descriptionStyle: {
    paddingLeft: 16,
    color: 'white',
  },
  divider: {
    backgroundColor: 'gray',
  },
});
