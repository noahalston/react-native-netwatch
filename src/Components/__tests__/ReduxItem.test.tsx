import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { ReduxItem, IProps } from '../ReduxItem';
import ReduxAction from '../../Core/Objects/ReduxAction';

describe('ReduxItem test suite', () => {
  let component: ShallowWrapper;
  let props: IProps;

  it('should render properly', () => {
    givenProps();
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  // GIVEN
  function givenComponent() {
    component = shallow(<ReduxItem {...props} />);
  }

  function givenProps() {
    props = {
      item: mockAction,
    };
  }

  const mockAction: ReduxAction = {
    _id: 73,
    startTime: 100,
    type: 'REDUX',
    action: { type: '__ERROR:UNDEFINED__', action: '' },
  };
});
