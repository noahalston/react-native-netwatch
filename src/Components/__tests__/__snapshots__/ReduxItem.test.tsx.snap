// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReduxItem test suite should render properly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#757575",
      "borderBottomColor": "#bdbdbd",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "height": 50,
    }
  }
>
  <Status
    backgroundColor="#64b5f6"
    item={
      Object {
        "_id": 73,
        "action": Object {
          "action": "",
          "type": "__ERROR:UNDEFINED__",
        },
        "startTime": 100,
        "type": "REDUX",
      }
    }
    subText="-"
    text="ACT"
  />
  <Component
    style={
      Object {
        "flex": 1,
        "paddingRight": 8,
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Component
        numberOfLines={1}
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
            "paddingLeft": 16,
          }
        }
      >
        __ERROR:UNDEFINED__
      </Component>
      <Component
        style={
          Object {
            "color": "white",
          }
        }
      >
        00:00:00
      </Component>
    </Component>
    <Component
      numberOfLines={1}
      style={
        Object {
          "color": "white",
          "paddingLeft": 16,
        }
      }
    >
      Redux action
    </Component>
  </Component>
</Component>
`;
